### ISSUE TITLE
modfftw contiguous pointer error with gfortran 8.5.0

### ISSUE THREAD

**Issue by fjansson** [dalesteam/dales#78]
gfortran 8.5.0 reports an error `Assignment to contiguous pointer from non-contiguous target` in modfftw.f90,
https://github.com/dalesteam/dales/blob/1d63af5d6fefbe2915be67e3be1e094f0634acb3/src/modfftw.f90#L165

`p210_flat` is declared contiguous, `fptr` is not. I haven't seen any other compilers complaining about this.
It seems to me it would be safe to declare also fptr as contiguous, because it comes from 
```
ptr = fftw_alloc_real(sz)
call c_f_pointer(ptr, fptr, (/sz/))
```
which should be a contiguous array. @jiskattema what do you think?
I tested making fptr contiguous, and at least it compiles then.

The compiler output:
```
/home/fjansson/dales/src/modfftw.f90:165:34:

     p210_flat(1:itot*jmax*konx)=> fptr(1:itot*jmax*konx)
                                  1
Error: Assignment to contiguous pointer from non-contiguous target at (1)
/home/fjansson/dales/src/modfftw.f90:167:34:

     p201_flat(1:itot*konx*iony)=> fptr(1:itot*konx*iony)
                                  1
Error: Assignment to contiguous pointer from non-contiguous target at (1)
/home/fjansson/dales/src/modfftw.f90:178:6:

       p210_flat,  &       ! array; location of transform k is: in + k * idist
      1
```

**Comment by jiskattema** [comment]
I added ``contiguous``` in the hope the compiler could make stuff a bit faster. I apparently missed ```fptr```. 

>  It seems to me it would be safe to declare also fptr as contiguous.

Sounds good to me too, If it compiles, and the output looks good, go ahead.

**Comment by fjansson** [comment]
Thanks @jiskattema !  

gfortran 8 apparently has a bug with handling contiguous, noted here: https://github.com/fabm-model/fabm/wiki/Building-and-installing#supported-compilers. Perhaps that's the reason other compilers don't complain. I will try adding contiguous also on fptr, since that fixes compilation on gfortran 8.

Otherwise a workaround is to not use FFTW with gfortran 8.

**Comment by fjansson** [comment]
Fixed with 731c2eacbbb6c1 released in v4.4.1.
